### PR TITLE
Add support for codeowners to repo

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
Add support for codeowners to repo
 
### Issues Resolved
Progress on:
[Meta issue](https://github.com/opensearch-project/project-meta/issues/31)
 
### Check List
- [ X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by CEHENKLE — APPROVED** [review]

**Review by CEHENKLE — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Comment by CEHENKLE** [comment]
@opensearch-project/opensearch-dashboards-core  I've updated the branch rules to use this codeowner file. What do you want to set the number of approvals required to merge a PRs as? 1 or 2?

At some point we may require consistency, but I don't think the direction is clear. So since I'm having to go in and update the rules, if you have a preference, LMK.

**Comment by tmarkley** [comment]
@CEHENKLE my vote is for 1 to maintain velocity since our core team is fairly small.

**Pull request merged**
